### PR TITLE
Influxdb cluster

### PR DESCRIPTION
**Pull request by AvitalFineRedis**

### CODE REVIEW

File: ansible/roles/influxdb/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- name: Generate configuration files
+  template:
+    owner: root
+    group: root
+    mode: 0644
+    dest: /etc/influxdb/influxdb.conf
+    src: influxdb.conf
+  register: influxconf
+
+- name:
```
Comment (chayim) [inline ansible/roles/influxdb/tasks/main.yml]: Missing name.

File: ansible/roles/influxdb/templates/influxdb.conf [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+###
+### Controls the parameters for the Raft consensus group that stores metadata
+### about the InfluxDB cluster.
+###
+
+[meta]
+  # Where the metadata/raft database is stored
+  {% if influxdb_inmemory %}dir = "/dev/shm/influxdb/meta"{% else %}dir = "/var/lib/influxdb/meta"{% endif %}
+
+  enabled = true
```
Comment (chayim) [inline ansible/roles/influxdb/templates/influxdb.conf:30]: Let's have these three items occur, only when building the cluster, and not set them by default.

File: ansible/roles/influxdb/templates/influxdb.conf [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+  # enabled = true
+
+  # Determines whether the Flux query endpoint is enabled.
+  # flux-enabled = false
+
+  # Determines whether the Flux query logging is enabled.
+  # flux-log-enabled = false
+
+  # The bind address used by the HTTP service.
+  bind-address = ":8086"
```
Comment (chayim) [inline ansible/roles/influxdb/templates/influxdb.conf]: Same as above - this should be set based on our specifically building slave nodes for the cluster.

File: ansible/roles/influxdb/templates/influxdefault [hunk 4]
```diff
@@ -0,0 +1 @@
+INFLUXD_OPTS="-join <IP1>:8091"
```
Comment (chayim) [inline ansible/roles/influxdb/templates/influxdefault]: In theory this should work: {{ansible_default_ipv4.address|default(ansible_all_ipv4_addresses[0])}}

Can you try testing this through, and see if the IP address if fulled in? <IP1> would be replaced with the item between the braces.

Alterantively "{{ ansible_default_ipv4.address }}" is supposed to work in newer versions of ansible.

**Review by chayim — CHANGES_REQUESTED** [review]

File: ansible/roles/influxdb/tasks/main.yml (hunk 1 above)
Comment (AvitalFineRedis) [reply ansible/roles/influxdb/tasks/main.yml]: updated th PR

**Pull request merged**